sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +32,112 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return m_response;
+}
+
+CURLcode CurlClient::perform()
+{
+  // init instance state
+  m_response = azure::core::http::Response();
+  m_firstHeader = true;
+
+  auto settingUp = setUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = setHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = setWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static void parseAndSetFirstHeader(std::string const& header, Response& response)
+{
+  // set response code, http version and reason phrase (i.e. HTTP/1.1 200 OK)
+  auto start = std::find(header.begin(), header.end(), '/');
+  start++; // skip symbol
+  auto end = std::find(start, header.end(), '.');
+  auto mayorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of minor version
+  end = std::find(start, header.end(), ' ');
+  auto minorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of status code
+  end = std::find(start, header.end(), ' ');
+  auto statusCode = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of reason phrase
+  auto reasonPhrase = std::string(start, header.end() - 2); // remove \r and \n from the end
+
+  response.setVersion(mayorVersion, minorVersion);
```

Comment:
Here's where you ctor a Response object passing all these values to the ctor and set the this's response field